edb/pgsql/compiler/clauses.py

```diff
@@ -52,7 +52,14 @@ def get_volatility_ref(
     if not ref:
         rvar = relctx.maybe_get_path_rvar(
             stmt, path_id, aspect='value', ctx=ctx)
-        if rvar and isinstance(rvar.query, pgast.ReturningQuery):
+        if (
+            rvar
+            and isinstance(rvar.query, pgast.ReturningQuery)
+            # XXX: Is this legit? Otherwise ran into trouble with
```

Comment:
I'm not 100% sure here, feels dangerous to me.  A cleaner solution would be to inject a tuple of `(row_number(), relation_number)` into every member of the union op, where `relation_number` is just an index of the relation or something.

Reply:
I wound up handling this by specifically exempting expanded inhviews, which we know will be safe to just use the value for, and I didn't want to have the volatility ref be substantially different in expanded mode.

I /think/ that other unions shouldn't be able to sneak in here (they'll typically be wrapped in another select), but if they do we'll fail instead of doing the wrong thing. If we encounter that we can do your suggestion, but I don't want to implement it without a test case, and I don't want to generate that code for these expanded inhviews